src/theory/bags/inference_generator.h

```diff
@@ -170,6 +176,37 @@ class InferenceGenerator
    * where skolem is a fresh variable equals (bag.duplicate_removal A)
    */
   InferInfo duplicateRemoval(Node n, Node e);
+  /**
+   * @param cardTerm a term of the form (bag.card A) where A has type (Bag E)
+   * @param n is (as bag.empty (Bag E))
+   * @return an inference that represents the following implication
+   * (=> (= A (as bag.empty (Bag E)))
+   *     (= (bag.card A) 0))
+   */
+  InferInfo cardEmpty(const std::pair<Node, Node>& pair, Node n);
+  /**
+   * @param cardTerm a term of the form (bag.card A) where A has type (Bag E)
+   * @param n is a node of the form (bag x c) of type (Bag E)
+   * @return an inference that represents the following implication
+   * (=>
+   *     (and (= A (bag x c)) (>= 0 c))
+   *     (= (bag.card A) c))
+   */
+  InferInfo cardBagMake(const std::pair<Node, Node>& pair, Node n);
+  /**
+   * @param premise a boolean node
```

Comment:
This parameter seems vague, what guarantees do we have on premise?

Reply:
I added a comment saying the premise explains why parent equals the disjoint union of its children. The premise can actually be literal true after rewriting when we have 
`(= (bag.union_disjoint A B) (bag.union_disjoint A B))`
But in general, we would have this equality `(= bagRep (bag.union_disjoint A B))` from the equality engine which we need as a premise to the implication in the generated inference.